pkg/liquidity-mining/test/GaugeMaxRelativeWeight.test.ts

```diff
@@ -0,0 +1,213 @@
+            expect(await gauge.max_relative_weight()).to.be.eq(defaultMaxRelativeWeight);
+            await setNewMaxRelativeWeight(newMaxRelativeWeight);
+            expect(await gauge.max_relative_weight()).to.be.eq(newMaxRelativeWeight);
+          });
+
+          it('emits an event', async () => {
+            const gaugeMaxRelativeWeightChangedInterface = new ethers.utils.Interface([
+              'event MaxRelativeWeightChanged(uint256 indexed new_max_relative_weight)',
+            ]);
+            const tx = await setNewMaxRelativeWeight(newMaxRelativeWeight);
+            expectEvent.inIndirectReceipt(
+              await tx.wait(),
+              gaugeMaxRelativeWeightChangedInterface,
+              'MaxRelativeWeightChanged',
+              { new_max_relative_weight: newMaxRelativeWeight }
+            );
+          });
+        });
+
+        context('above the allowed range', () => {
+          it('reverts', async () => {
+            await expect(setNewMaxRelativeWeight(aboveAbsoluteMaxRelativeWeight)).to.be.revertedWith(
+              'Max relative weight exceeds allowed absolute maximum'
+            );
+          });
+        });
+      });
+    });
+
+    describe('get_capped_relative_weight', () => {
+      const gaugeControllerWeight = fp(0.7);
+      const maxRelativeWeightAboveControllerWeight = gaugeControllerWeight.add(1);
+      const maxRelativeWeightBelowControllerWeight = gaugeControllerWeight.sub(1);
+      // The timestamp parameter is being ignored in the mock gauge controller.
+      const anyTimestamp = bn(1234);
+
+      context('when the gauge is not added to the gauge controller', () => {
+        it('returns 0', async () => {
+          expect(await gauge.get_capped_relative_weight(anyTimestamp)).to.be.eq(0);
+        });
+      });
+
+      context('when the gauge is added to the gauge controller', () => {
+        sharedBeforeEach('add gauge to GaugeController', async () => {
+          await gaugeController.add_gauge_with_weight(gauge.address, 0, gaugeControllerWeight);
+        });
+
+        const itChecksMaxRelativeWeightHasNoEffect = (maxRelativeWeightUnderTest: BigNumber, condition: string) => {
+          context(`when max relative weight ${condition} gauge controller relative weight for the gauge`, () => {
+            let uncappedRelativeWeight: BigNumber;
+            sharedBeforeEach(
+              `set gauge max relative weight ${condition} gauge controller relative weight for the gauge`,
+              async () => {
+                await setNewMaxRelativeWeight(maxRelativeWeightUnderTest);
+                uncappedRelativeWeight = await gaugeController.gauge_relative_weight(gauge.address, anyTimestamp);
+              }
+            );
+
+            it('verifies that max relative weight >= gauge controller relative weight for the gauge', async () => {
+              expect(await gauge.max_relative_weight()).to.be.gte(uncappedRelativeWeight);
+            });
```

Comment:
This is not a test - it's a test on the tests themselves, to make sure you didn't screw up the setup, a sanity check. 

If you want to include it (which is not a bad idea), don't make an `it` out of it, but instead add it at the end of the `sharedBeforeEach` (the setup you're testing)

Reply:
+1!